### DISCUSSION TITLE
How to use metrics with multi labels

### DISCUSSION THREAD

**Discussion by FrancescoSaverioZuppichini** [Lightning-AI/torchmetrics#526] [Classification] [answered]
Hi guys,

I hope you are doing great! I was wondering how to correctly use a metric, e.g. Accuracy, with multi labels. For example

```python

num_classes = 4

pred = torch.tensor([0, 0, 0, 0])
target = torch.tensor([0, 1, 0, 1])

acc = M.Accuracy(num_classes=num_classes)
print(acc(pred, target))
```

Where `target` is a one-hot vector and if `target[i] == 1` it means that we have class `i` in the target. In this case, accuracy doesn't work, since it only compares the raw data inside it. 

Thanks!

**Comment by SkafteNicki** [comment, marked as answer]
For multi-labels we assume that you are providing data as `[B,L]` where `B` is the batch size and `L` is the number of labels (`num_classes` in your code). You are therefore missing the batch dimension, which is 1 in your case. A simple `unsqueeze` should be enough:
```python
pred = torch.tensor([0, 0, 0, 0]).unsqueeze(0)
target = torch.tensor([0, 1, 0, 1]).unsqueeze(0)
```
